src/main/java/org/opensearch/geospatial/ip2geo/common/DatasourceFacade.java

```diff
@@ -120,4 +131,77 @@ public void onFailure(final Exception e) {
             }
         });
     }
+
+    /**
+     * Get datasources from an index {@code DatasourceExtension.JOB_INDEX_NAME}
+     * @param names the array of datasource names
+     * @param actionListener the action listener
+     */
+    public void getDatasources(final String[] names, final ActionListener<List<Datasource>> actionListener) {
+        client.prepareMultiGet()
+            .add(DatasourceExtension.JOB_INDEX_NAME, names)
+            .execute(newActionListener(MultiGetResponse.class, actionListener));
+    }
+
+    /**
+     * Get all datasources from an index {@code DatasourceExtension.JOB_INDEX_NAME}
+     * @param actionListener the action listener
+     */
+    public void getAllDatasources(final ActionListener<List<Datasource>> actionListener) {
+        client.prepareSearch(DatasourceExtension.JOB_INDEX_NAME)
+            .setQuery(QueryBuilders.matchAllQuery())
+            .execute(newActionListener(SearchResponse.class, actionListener));
```

Comment:
can we add a size parameter as 100 or 1k here just to be on safe side if number of datasource values go above 10 which is default size value.